lib/active_merchant/billing/gateways/nuvei.rb

```diff
@@ -150,6 +151,53 @@ def add_address(post, payment, options)
         }.compact
       end
 
+      def execute_3ds_flow(money, payment, transaction_type, options = {})
```

Comment:
💬 Comment:

A possible refactor to be more explicit on the code flow:

```ruby
      def execute_3ds_flow(money, payment, transaction_type, options = {})
        post = {}
        build_payment_post(post, money, payment, options)
        post_3ds = post.dup

        MultiResponse.run do |r|
          r.process { commit(:init_payment, post) }
          r.process do
            post_3ds.merge({ relatedTransactionId: r.authorization, transactionType: transaction_type })
            add_3ds_data(post_3ds, options.merge({ version: r.params.dig('paymentOption', 'card', 'threeD', 'version') }))
            commit(:purchase, post_3ds)
          end
        end
      end

```

Reply:
💬 Comment

Take into consideration the `not enrolled` case when CC is no available for 3DS.